controllers/harborconfiguration_controller.go

```diff
@@ -390,19 +390,59 @@ func getConcreteHarborType(ctx context.Context, crdClient dynamic.ResourceInterf
 }
 
 func deleteAll(ctx context.Context, harborConfiguration harborconfigurationv1alpha1.HarborConfiguration, client *apiv2.RESTClient) (ctrl.Result, error) {
-	err := client.DeleteRegistryByID(ctx, harborConfiguration.Status.RegistryId)
-	if err != nil {
+	deleteReplicationRule(ctx, harborConfiguration, client)
+	deleteProject(ctx, harborConfiguration, client)
+	deleteRegistry(ctx, harborConfiguration, client)
+	return ctrl.Result{}, nil
+}
+
+func deleteReplicationRule(ctx context.Context, harborConfiguration harborconfigurationv1alpha1.HarborConfiguration, client *apiv2.RESTClient) (ctrl.Result, error) {
+	replicationFound, err := client.GetReplicationPolicyByName(ctx, harborConfiguration.Spec.Replication.Name)
+	if errors.Is(err, &harborerrors.ErrNotFound{}) {
 		return ctrl.Result{}, err
+	} else if err != nil {
+		return ctrl.Result{}, err
+	} else {
+		err = client.DeleteReplicationPolicyByID(ctx, replicationFound.ID)
+		if err != nil {
+			return ctrl.Result{}, err
```

Comment:
ditto

Reply:
I've removed the redundant else if and else statements, the reason I had them was because initially I was returning different logs for each. Returning the error should show it in the stack trace if one occurs.